db/migrate/20230314192630_update_bad_distribution_issued_ats.rb

```diff
@@ -0,0 +1,9 @@
+class UpdateBadDistributionIssuedAts < ActiveRecord::Migration[7.0]
+  def up
+    Distribution.unscoped.where("issued_at <  ?","2000-01-01").find_each do |dist|
+      # Using update_attribute, rather than update! because some of the distributions
+      # will not validate for other reasons (storage locations missing items)
+      dist.update_attribute(:issued_at, dist.created_at)
```

Comment:
You can actually do this in a single SQL call which might be less error prone:

```
Distribution.unscoped.where("issued_at <  ?","2000-01-01").
  update_all('issued_at=created_at, updated_at=NOW()')
```

Reply:
Is the intent of updated at when the user last touched it, or when it was changed in the db?   Because if the former, we shouldn't touch it.     I know we've had some reports that have been based on the updated_at -- I think in error, but I am feeling a bit cautious about touching it here.

Reply:
I'm pretty sure from my inventory drift investigations that there are some occasions when we are *not* touching it in error also... (sigh)  We should, perhaps, have someone dig into that.

Reply:
Yes, `updated_at` should, by convention, always be when the _record_ was updated for any normal reason even if not caused by a human interaction. So like a background job or something should update it. Probably a good best-practice to update it as part of this change.

Reply:
Yeah --That's what I would have assumed -- but I noticed that we aren't in some cases when I was working on the inventory drift problem.  So I'm concerned that we may be depending on it not changing.   I'm now also wondering how this will should interact with the historical inventory on the storage location.  Enough that I'm going to throw a WIP onto this until I can check *that* out.

Reply:
I think we do need to noodle the historical inventory and distributions a bit more.  In most cases, though, it's the created_at date that's going to be matching up to the inventory,  not the updated_at.  The exception will be any corrections -- which I presume would see the inventory nudged appropriately at the updated_at date.

Reply:
All of which is to say we likely won't be making any issues worse by setting the updated to now

Reply:
So this change should still happen then?